docs/site/migration/express-middleware.md

```diff
@@ -6,8 +6,251 @@ sidebar: lb4_sidebar
 permalink: /doc/en/lb4/migration-express-middleware.html
 ---
 
+This is a continuation of the
+"[Mounting a LoopBack 3 application](https://loopback.io/doc/en/lb4/migration-mounting-lb3app.html)"
+guide. The instructions will continue with the assumption that you have a base
+app as instrcuted in
+https://github.com/strongloop/loopback-next/tree/master/examples/lb3-application#tutorial.
+
+Now that you have a LoopBack 3 (LB3) app mounted on a LoopBack 4 (LB4) app, it
+is time to further optimize the setup so that only the bare neccessary artifacts
+from the LoopBack 3 app remains, that includes moving all the middleware to a
+common location so that they are shared by both the LoopBack 3 and LoopBack 4
+apps.
+
+First off, edit the LB3 app's `config.json` file.
+
+Remove these properties, they are not required anymore:
+
+```json
+"restApiRoot": "/api",
+"host": "0.0.0.0",
+"port": 3000,
+```
+
+And then add `"handleUnknownPaths": false` to the `rest` property, this will
+prevent the LB3 REST api from sending a 404 response for requests it cannot
+handle.
+
+The `config.json` file should now look like this:
+
+{% include code-caption.html content="lb3app/server/config.json" %}
+
+```json
+{
+  "remoting": {
+    "context": false,
+    "rest": {
+      "handleErrors": false,
+      "handleUnknownPaths": false,
+      "normalizeHttpPath": false,
+      "xml": false
+    },
+    "json": {
+      "strict": false,
+      "limit": "100kb"
+    },
+    "urlencoded": {
+      "extended": true,
+      "limit": "100kb"
+    },
+    "cors": false
+  }
+}
+```
+
+We will be using a base Express app for mounting the LB4 app. Notice how we are
+loading the middleware from the LB3 in this app. These middleware will be shared
+by both LB3 and LB4 apps. We are not adding the `loopback.rest` middleware, it
+will be loaded by the `Lb3AppBooterComponent`.
+
+{% include code-caption.html content="lb3app/server.ts" %}
+
+```ts
+import {ApplicationConfig} from '@loopback/core';
+import * as express from 'express';
+import {Request, Response} from 'express';
+import * as http from 'http';
+import pEvent from 'p-event';
+import * as path from 'path';
+import {CoffeeShopApplication} from './application';
+
+const loopback = require('loopback');
+const compression = require('compression');
+const cors = require('cors');
+const helmet = require('helmet');
+
+export class ExpressServer {
+  private app: express.Application;
+  public readonly lbApp: CoffeeShopApplication;
+  private server?: http.Server;
+
+  constructor(options: ApplicationConfig = {}) {
+    this.app = express();
+    this.lbApp = new CoffeeShopApplication(options);
+
+    // Middleware for LB3
+    this.app.use(loopback.favicon());
+    this.app.use(compression());
+    this.app.use(cors());
+    this.app.use(helmet());
```

Comment:
I am concerned that this example shows only middleware from our CoffeeShop example application. Please find a way how to provide more generic instructions and explain users how to migrate `server/middleware.json` entries like `loopback#favicon` to TypeScript code using `this.app.use()`.

The code snippets should be examples illustrating the instructions written in plain English, not the instructions themselves.

Reply:
Added the detailed, starts at "Migrate the LB3 app's middleware from its ..."